### PR TITLE
Doc for adding a new part to PrjXray.

### PR DESCRIPTION
**Pull request by tcal-x**
Signed-off-by: Tim Callahan <tcal@google.com>

### CODE REVIEW

File: NEWPART.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
+
```
Comment (litghost) [inline NEWPART.md:1]: Should be in `docs/`?

File: NEWPART.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+```
+make -j32 MAX_VIVADO_PROCESS=32 db-part-only-artix7_100t
+```
+
+* It should run fuzzers 000, 001, 005, 072, 073, 074, and 075.
+                
+* 005 will take a long time unless you throw a lot of cores at it.  32-64 cores suggested.
+
+* You _may_ get a spurious failure, since the parallel runs may violate ordering dependencies among the fuzzers.  If this happens, clean the fuzzer that failed (e.g.  `make -C fuzzers/005-tilegrid clean`), and then rerun the top-level make.
```
Comment (litghost) [inline NEWPART.md:74]: There shouldn't be ordering failing in 005?

File: NEWPART.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+
+
+### Step 2 ###
+
+Add a new settings file.  Usually you will start with an existing settings file and modify it.
+Assuming you're in prjxray/,
+
+```
+cp settings/artix7_200t.sh settings/artix7_100t.sh
```
Comment (litghost) [inline NEWPART.md:34]: Please change to two instances of what needs to be changed.  One that makes clear the old and new, and an example.  So for example this block would be:

```
cp settings/<baseline settings>.sh settings/<new settings>.sh
git add settings/<new settings>.sh
```

Example:

```
cp settings/artix7_200t.sh settings/artix7_100t.sh
git add settings/artix7_100t.sh
```

File: NEWPART.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+```
+
+### Step 3 ###
+
+Edit the top Makefile
+
+* Add the new part to the correct list.   This is the basename of the new settings file that you just created. In this case, we added `artix7_100t`:
+
+```
+ARTIX_PARTS=artix7_200t artix7_100t
```
Comment (litghost) [inline NEWPART.md:58]: Add a github permalink to the lines you are talking about here to the doc.  This step is a little hard to follow unless you first read the Makefile.  Maybe setup what this line is doing, and why you need to change it?

File: NEWPART.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+```
+
+
+### Step 4 ###
+
+Make the following target, adapted for your new part, and setting the parallelism factor appropriate for your machine:
```
Comment (litghost) [inline NEWPART.md:64]: Explain what this is going to do.  The is also a good example of where the template comment from above is important.  In this case, the template version is:

```
make -j32 MAX_VIVADO_PROCESS=32 db-part-only-<new settings>
```

And make clear that, the name of the settings sh files and the change to the Makefile and the name here all need to be the same.

File: NEWPART.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+    * Run `python3 ../analyze_errors.py --output_ignore_list  > new-ignored`
+    * Inspect and compare against existing ignored wire files
+    * IF IT LOOKS GOOD, copy it to an appropriately-named file: `cp new-ignored ../ignored_wires/artix7/xc7a100tfgg676-1_ignored_wires.txt`
+    * Add it: `git add ../ignored_wires/artix7/xc7a100tfgg676-1_ignored_wires.txt`
+
+* Return to prjxray/ directory, and clean up 074 to prepare for the rerun: `make -C fuzzers/074-dump-all clean`
+
+* Rerun the top make command, `make -j32 MAX_VIVADO_PROCESS=32 db-part-only-artix7_100t`
+
+### Step 5 ###
```
Comment (litghost) [inline NEWPART.md:87]: When this is committed, add a comment to https://github.com/SymbiFlow/prjxray/issues/1476 to remove this step once we have updated the part processing stuff.  The instructions here also need to be updated as part of https://github.com/SymbiFlow/prjxray/issues/1475 too

File: NEWPART.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+```
+git push origin master  // or whatever branch you're using
+```
+
+Then make a pull request.  
+Navigate to the GitHub page for your fork, and click the "New pull request" button.
+Just making the request will kick off Kokoro runs.
+
+
+# BUT WAIT, WHAT ABOUT THE DB #
```
Comment (litghost) [inline NEWPART.md:171]: This section could use a tidy.

File: NEWPART.md [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+git add Makefile settings/artix7_100t.sh
+git status
+git commit --signoff 
+```
+
+### Step 8 ###
+
+Push to GitHub:
+```
+git push origin master  // or whatever branch you're using
```
Comment (litghost) [inline NEWPART.md:163]: I generally recommend that users create a new branch when submitting PRs with a useful branch name.

File: NEWPART.md [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+_If you are **just** adding a new package for a device that is already handled, you can skip Steps 2..4!_
+
+
+### Step 0 ###
+Fork a copy of https://github.com/SymbiFlow/prjxray on GitHub (go to the page, click "Fork" button, select your own workspace).
+
+Clone your fork (replace "tcal-x" with your GitHub ID):
+
+```
+git clone git@github.com:tcal-x/prjxray.git
```
Comment (mkurc-ant) [inline NEWPART.md:16]: You may just point to the SymbiFlow/prjxray. There will be no need for instructing a user to replace your github id.

**Review by mkurc-ant — COMMENTED** [review]
You have written this doc more like a report of what you did and how. In my opinion an instruction should be written in a more impersonal way.

File: NEWPART.md (hunk 1 above)
Comment (tcal-x) [reply NEWPART.md:1]: Yeah, makes sense.   The question is, /only/ in docs as .rst, or also have an .md version?   I noticed that the getting started readme exists in both formats.

File: NEWPART.md (hunk 2 above)
Comment (tcal-x) [reply NEWPART.md:74]: 001 must produce the parts yaml for 005.    But yeah, now when I run the target `db-part-only-<part>`, I see that 005 waits for 001 to finish.  Do you think that respecting the dependency was added recently, or did I just see some fluke, maybe due to a restart, where 005 started when the parts yaml wasn't there?   If this should never happen, I'll take out the section.

Comment (litghost) [reply NEWPART.md:74]: I believe there shouldn't be an issue here, and if there was we'd want to file a bug, replicate the issue and then fix it.

File: NEWPART.md (hunk 8 above)
Comment (tcal-x) [reply NEWPART.md:163]: I'm updating the instructions to create a branch with a useful name.

File: NEWPART.md (hunk 9 above)
Comment (tcal-x) [reply NEWPART.md]: I've replaced "tcal-x" with "<yourGithubId>", since I don't want them to start hacking directly on their clone of SymbiFlow's repo.   Yes, if they know enough git they will know how to update things so they can commit to their own fork later, but it seems simpler if they start with their own fork...

File: NEWPART.md (hunk 1 above)
Comment (tcal-x) [reply NEWPART.md:1]: I'll ask for re-reviews after updating the content here, and then when it looks good, I'll convert it to an .rst file under docs/.

File: NEWPART.md (hunk 6 above)
Comment (tcal-x) [reply NEWPART.md:114]: Yes, good catch.   I'll leave this comment unresolved, and add a comment to those PRs when this doc's final location is determined.

Comment (tcal-x) [reply NEWPART.md:114]: I also need to check if https://github.com/SymbiFlow/symbiflow-arch-defs/pull/1791 affects my instructions (under "Database Updates")

File: NEWPART.md (hunk 1 above)
Comment (tcal-x) [reply NEWPART.md:2]: @mkurc-ant , @litghost , the new doc (docs/newpart.rst) is on readthedocs, ready for review.   Ignore the NEWPART.md file; I'll delete it before merging.   Start here:  [https://symbiflow--1486.org.readthedocs.build/projects/prjxray/en/1486/db_dev_process/index.html](https://symbiflow--1486.org.readthedocs.build/projects/prjxray/en/1486/db_dev_process/index.html) (to see where I've put the new page in the hierarchy), then click on "Guide to adding...".

File: README.md [hunk 10]
```diff
@@ -200,6 +200,10 @@ We have also started experimenting with the Kintex-7 parts.
 The aim is to eventually document all parts in the Xilinx 7-series FPGAs but we
 can not do this alone, **we need your help**!
 
+## Adding a new part to an existing family
+
+We have written a [detailed guide](./NEWPART.md) that walks through the process
```
Comment (litghost) [inline README.md:205]: This should reference the RST version of the doc?

File: NEWPART.md (hunk 1 above)
Comment (litghost) [reply NEWPART.md:2]: Is it time to delete this file?

File: docs/db_dev_process/newpart.rst [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+   git clone git@github.com:<yourUserID>/prjxray.git
+   cd prjxray
+   git checkout -b <new_branch_name>
+
+Step 1
+~~~~~~
+
+Follow the Project X-Ray developer setup instructions in the
+`documentation <https://symbiflow.readthedocs.io/en/latest/prjxray/docs/db_dev_process/readme.html>`__,
+up through Step 7, Option 1. That is, source the
```
Comment (litghost) [inline docs/db_dev_process/newpart.rst:48]: I don't believe it sources this particular script, it just invokes it.

**Review by litghost — COMMENTED** [review]
Overall looks great!  Let's cleanup the handful of comments, and get this merged!

**Review by litghost — APPROVED** [review]
LGTM
